src/NoteWorkspace.ts

```diff
@@ -171,6 +171,8 @@ export class NoteWorkspace {
   static normalizeNoteNameForFuzzyMatchText(noteName: string): string {
     // remove the brackets:
     let n = noteName.replace(/[\[\]]/g, '');
+    // remove the potential wikilink:
+    n = n.replace(/\|[^\\\[]+$/, '');
```

Comment:
OK, so am I correct in understanding the way this works is that the pipe and succeeding characters are stripped, such that:

```[[test-file.md|Some Test File]]``

Will resolve to the same thing as:

```[[test-file.md]]``

I can see this change affecting the following behaviors:

- `editor.action.revealDefinition` - *Go To Definition* on `[[test-file.md|Some Test]]` should open `test-file.md` (if `test-file.md` exists on file system)
- `createNoteOnGoToDefinitionWhenMissing` - `editor.action.revealDefinition`(missing *Go To Definition*) on `[[test-file.md|Some Test]]` should CREATE and open `test-file.md` (if `test-file.md` does NOT exist on file system)
- `editor.action.peekDefinition` - should preview `test-file.md` (if `test-file.md` does exists on file system)
- references / backlinks `[[test-file.md|Some Test]]` should register as a reference / backlink to `test-file.md`

### questions

- Does your PR address all of these behaviors, or just some of them?
- Can you please add some test cases involving piped links in https://github.com/kortina/vscode-markdown-notes/blob/master/src/test/jest/extension.test.ts
- Can you feature flag this with a new config setting, `vscodeMarkdownNotes.allowPipedWikiLinks`, where if this is true we strip the pipe and succeeding characters, otherwise treat the pipe as any other special character that gets replaced with the `slugifyCharacter`
- Perhaps we should make the pipe character itself configurable, defaulting it to "|" ? @lukesmurray suggested making this configurable elsewhere.
- Do Obsidian, Roam Research, Foam, and other projects all treat the characters before the pipe as the file and the characters after as the link text (or does anyone do the inverse)? Seems like your PR follows the [mediawiki style](https://www.mediawiki.org/wiki/Help:Links/sco) and I just want to make sure this how it works for other projects.

Thank you for the PR! I am excited to add this functionality, as several other users of the extension have asked for it.

Lmk if you have any questions about how to proceed with the PR by addressing the feedback.

---

This PR will address #56 and #66